src/verifier.rs

```diff
@@ -391,22 +391,27 @@ impl Verifier for RequisiteVerifier {
                 ebpf::JSLE_IMM   => { check_jmp_offset(prog, insn_ptr, &function_range)?; },
                 ebpf::JSLE_REG   => { check_jmp_offset(prog, insn_ptr, &function_range)?; },
                 ebpf::CALL_IMM   if sbpf_version.static_syscalls() => {
-                    let target_pc = sbpf_version.calculate_call_imm_target_pc(insn_ptr, insn.imm);
-                    check_call_target(
-                        target_pc,
-                        function_registry,
-                        VerifierError::InvalidFunction(target_pc as usize)
-                    )?;
+                    if sbpf_version.stricter_controlflow() {
+                        let target_pc = sbpf_version.calculate_call_imm_target_pc(insn_ptr, insn.imm);
+                        check_call_target(
+                            target_pc,
+                            function_registry,
+                            VerifierError::InvalidFunction(target_pc as usize)
+                        )?;
+                    }
                 },
                 ebpf::CALL_IMM   => {},
                 ebpf::CALL_REG   => { check_callx_register(&insn, insn_ptr, sbpf_version)?; },
                 ebpf::EXIT       if !sbpf_version.static_syscalls()   => {},
                 ebpf::RETURN     if sbpf_version.static_syscalls()    => {},
                 ebpf::SYSCALL    if sbpf_version.static_syscalls()    => {
-                    check_call_target(
-                        insn.imm as u32,
-                        syscall_registry,
-                        VerifierError::InvalidSyscall(insn.imm as u32))?;
+                    if sbpf_version.stricter_controlflow() {
```

Comment:
Does this one need two gates?

Reply:
One to make it a valid opcode and one for the verification rule. These are two different SIMDs but same SBPF version.

Reply:
The verification rule for static syscalls is in the static syscalls SIMD.